### DISCUSSION TITLE
How to generate self-contained bundle with rollup

### DISCUSSION THREAD

**Discussion by derolf** [mikro-orm/mikro-orm#2266] [Q&A] [answered]
This is a documentation request.

Currently, we use the TsMorphMetadataProvider and run the server with ts-node.

We are trying to create a production bundle with rollup and don’t want to ship the entire ts-morph and typescript codebase.

So, what is the right approach to generate the metadata during bundling and get rid of TsMorphMetadataProvider during production?

**Comment by B4nan** [comment, marked as answer]
This is a bit more complex thing, that might be too hard to achieve with current implementation, so I would suggest going with reflect metadata instead (and not using folder based discovery at all).

Some interesting reading regarding this topic:
https://github.com/mikro-orm/mikro-orm/discussions/2103
https://github.com/mikro-orm/mikro-orm/discussions/2169
